app/src/components/publish/SubmitSurvey.tsx

```diff
@@ -42,25 +35,30 @@ export const SurveySubmitFormYupSchema = yup.object().shape({
   attachments: yup.array()
 });
 
-const SubmitSurvey: React.FC<ISubmitSurvey> = (props) => {
+const SubmitSurvey: React.FC = () => {
   const biohubApi = useBiohubApi();
+  const surveyContext = useSurveyContext();
 
-  const { surveyDetails } = props;
+  const surveyDetails = surveyContext.surveyDataLoader.data;
 
   const observationDataLoader = useDataLoader((projectId: number, surveyId: number) =>
     biohubApi.observation.getObservationSubmission(projectId, surveyId)
   );
+
   useDataLoaderError(observationDataLoader, () => {
     return {
       dialogTitle: 'Error Loading Occurrence Details',
       dialogText:
         'An error has occurred while attempting to load occurrence details, please try again. If the error persists, please contact your system administrator.'
     };
   });
-  observationDataLoader.load(
-    surveyDetails.surveyData.survey_details.project_id,
-    surveyDetails.surveyData.survey_details.id
-  );
+
+  if (surveyDetails) {
+    observationDataLoader.load(
+      surveyDetails.surveyData.survey_details.project_id,
+      surveyDetails.surveyData.survey_details.id
```

Comment:
The project/survey ids could come from useParams as well (Not sure if that changes anything).

Reply:
My intent was basically to let the Survey Context be the source of truth for survey and project ID, rather than sometimes getting this info from the URL params and sometimes getting it from the survey context. But yes in this case, we could call `useParams` instead, sinec we're just getting the two IDs

Reply:
> My intent was basically to let the Survey Context be the source of truth for survey and project ID

Yeah thats fair. I think that is totally fine, as long as we are consistent everywhere.